client/components/domains/register-domain-step/index.jsx

```diff
@@ -541,10 +539,20 @@ const RegisterDomainStep = React.createClass( {
 				severity = 'info';
 				break;
 			case 'not_registrable':
-				if ( domain.indexOf( '.' ) ) {
-					message = this.translate( 'Sorry but %(domain)s cannot be registered on WordPress.com.', {
-						args: { domain: domain }
-					} );
+				const tldIndex = domain.lastIndexOf( '.' );
+				if ( tldIndex !== -1 ) {
+					message = this.translate(
+						'To use a domain ending with {{strong}}%(tld)s{{/strong}} on your site, ' +
+						'you can register it elsewhere and then add it here. {{a}}Learn more{{/a}}.',
+						{
```

Comment:
I don't think "map" is commonly understood. We use it a lot, but if it's not necessary, I think it's better to avoid it.

If they are already trying to add it here, this notice still works. The important part is just that they first register it elsewhere.

If you think it will help, we can make that even clearer:

> To use a domain ending with .nl to your site, you can register it elsewhere first and then add it here. Learn more.